crates/sui-framework/packages/move-stdlib/sources/vector.move

```diff
@@ -158,4 +158,149 @@ module std::vector {
         v.swap(i, last_idx);
         v.pop_back()
     }
+
+    // === Macros ===
+
+    /// Destroy the vector `v` by calling `f` on each element and then destroying the vector.
+    /// Does not preserve the order of elements in the vector (starts from the end of the vector).
+    public macro fun destroy<$T>($v: vector<$T>, $f: |$T|) {
+        let mut v = $v;
+        while (!v.is_empty()) {
+            $f(v.pop_back());
+        };
```

Comment:
Fine with this. I feel a lot less strongly about this than with `if`, but I am always partial to 
```suggestion
        while (!v.is_empty()) $f(v.pop_back());
```